tests/helpers/load_state_tests.py

```diff
@@ -325,24 +302,86 @@ def add_blocks_to_chain(
         load.state_transition(chain, block)
 
 
+# Functions that fetch individual test cases
+def load_json_fixture(test_file: str, network: str) -> Generator:
+    # Extract the pure basename of the file without the path to the file.
+    # Ex: Extract "world.json" from "path/to/file/world.json"
+    pure_test_file = os.path.basename(test_file)
+    # Extract the filename without the extension. Ex: Extract "world" from
+    # "world.json"
+    test_name = os.path.splitext(pure_test_file)[0]
+    with open(test_file, "r") as fp:
+        data = json.load(fp)
+
+        # Some newer test files have patterns like _d0g0v0_
+        # between test_name and network
+        keys_to_search = re.compile(
+            f"{re.escape(test_name)}.*{re.escape(network)}"
+        )
+        found_keys = list(filter(keys_to_search.match, data.keys()))
+
+        if not any(found_keys):
+            raise KeyError
+
+        for _key in found_keys:
+            yield {
+                "test_file": test_file,
+                "test_key": _key,
+                "test_data": data[_key],
+            }
+
+
 def fetch_state_test_files(
     test_dir: str,
-    slow_test_list: Tuple[str, ...],
-    incorrect_tests_list: Tuple[str, ...],
-    load: BaseLoad,
-) -> Generator[Union[str, ParameterSet], None, None]:
-    for _dir in os.listdir(test_dir):
-        test_file_path = os.path.join(test_dir, _dir)
-        for _file in os.listdir(test_file_path):
-            _test_file = os.path.join(_dir, _file)
-            if _test_file in incorrect_tests_list:
-                continue
-            elif _test_file in slow_test_list:
-                yield pytest.param(_test_file, marks=pytest.mark.slow)
-            else:
-                try:
-                    load_json_fixture(test_dir, _test_file, load)
-                    yield _test_file
-                except KeyError:
-                    # file doesn't contain tests for the given fork
-                    pass
+    network: str,
+    custom_file_list: Tuple[str, ...] = (),
```

Comment:
```suggestion
    only_in: Tuple[str, ...] = (),
```

I think this is more clear than `custom_file_list`, especially since these files are not appended to the discovered tests.

Reply:
Updated